core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3875,6 +3876,51 @@ class KafkaApis(val requestChannel: RequestChannel,
 
   }
 
+  def handleShareGroupDescribe(request: RequestChannel.Request): CompletableFuture[Unit] = {
+    val shareGroupDescribeRequest = request.body[ShareGroupDescribeRequest]
+
+    if (!config.isNewGroupCoordinatorEnabled) {
+      // The API is not supported by the "old" group coordinator (the default). If the
+      // new one is not enabled, we fail directly here.
+      requestHelper.sendMaybeThrottle(request, request.body[ShareGroupDescribeRequest].getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
+      CompletableFuture.completedFuture[Unit](())
+    } else {
+      val response = new ShareGroupDescribeResponseData()
+
+      val authorizedGroups = new ArrayBuffer[String]()
+      shareGroupDescribeRequest.data.groupIds.forEach { groupId =>
+        if (!authHelper.authorize(request.context, DESCRIBE, GROUP, groupId)) {
+          response.groups.add(new ShareGroupDescribeResponseData.DescribedGroup()
+            .setGroupId(groupId)
+            .setErrorCode(Errors.GROUP_AUTHORIZATION_FAILED.code)
+          )
+        } else {
+          authorizedGroups += groupId
+        }
+      }
+
+      groupCoordinator.shareGroupDescribe(
+        request.context,
+        authorizedGroups.asJava
+      ).handle[Unit] { (results, exception) =>
+        if (exception != null) {
+          requestHelper.sendMaybeThrottle(request, shareGroupDescribeRequest.getErrorResponse(exception))
+        } else {
+          if (response.groups.isEmpty) {
```

Comment:
Shouldn't it be `results.groups.isEmpty`? Isn't everytime this condition should be true, do we have a test case for else block?

Reply:
Seems like we do not have any tests for this class and others, please if you can add.

Reply:
> Shouldn't it be results.groups.isEmpty
The response object is being created above this block of code (line 3888)

It should be response itself. So we are updating the response object in this block. There is a small optimization being done where if the response.groups is empty at this point, we are setting reference to groups to that of result. `if response.groups.isempty then response.groups -> results.groups`, making it a O(1) time operation. Otherwise we are updating the list of groups already in response.groups which would be at best O(k) where k is number of groups to be added to existing list of groups.

Reply:
> Seems like we do not have any tests
Missed this - will rectify

Reply:
> It should be response itself

Seems right 👍